src/control/ros/Conversions.cpp

```diff
@@ -223,27 +239,44 @@ std::vector<const dart::dynamics::Joint*> findJointByName(
 //=============================================================================
 std::unique_ptr<SplineTrajectory> toSplineJointTrajectory(
     const std::shared_ptr<MetaSkeletonStateSpace>& space,
-    const trajectory_msgs::JointTrajectory& jointTrajectory)
+    const trajectory_msgs::JointTrajectory& jointTrajectory,
+    const Eigen::VectorXd& startPositions)
 {
   if (!space)
     throw std::invalid_argument{"StateSpace must be non-null."};
 
+  // Check that the number of joints specified in JointTrajectory or start
+  // position match the state space.
+  // 1. A trajectory must either specify as many joints as are in the space (no
+  //    padding) or fewer joints than are in the space (padding).
+  // 2. A start position must either have a size of 0 (no padding) or a size
+  //    matching the number of joints in the space (padding).
   const auto numControlledJoints = space->getNumSubspaces();
-  if (jointTrajectory.joint_names.size() != numControlledJoints)
+  const auto numTrajectoryJoints = jointTrajectory.joint_names.size();
+  if (numTrajectoryJoints > numControlledJoints ||
+      (startPositions.size() == 0 && numTrajectoryJoints != numControlledJoints))
   {
     std::stringstream message;
     message << "Incorrect number of joints: expected " << numControlledJoints
-            << ", got " << jointTrajectory.joint_names.size() << ".";
+            << ", got " << numTrajectoryJoints << ".";
+    throw std::invalid_argument{message.str()};
+  }
+  else if (startPositions.size() > 0 &&
+           static_cast<std::size_t>(startPositions.size()) != numControlledJoints)
+  {
+    std::stringstream message;
+    message << "Incorrect number of joints in configuration: expected "
+            << numControlledJoints << ", got " << startPositions.size() << ".";
     throw std::invalid_argument{message.str()};
   }
 
   // Check that the names in jointTrajectory are unique.
   std::vector<std::string> joint_names;
   joint_names.reserve(numControlledJoints);
-  for (size_t i = 0; i < numControlledJoints; ++i)
+  for (size_t i = 0; i < numTrajectoryJoints; ++i)
     joint_names.emplace_back(jointTrajectory.joint_names[i]);
```

Comment:
Unrelated nitpick: I think this vector copy can be simplified using the copy constructor as
```cpp
  std::vector<std::string> joint_names(
      jointTrajectory.joint_names.begin(), jointTrajectory.joint_names.end());
```